### DISCUSSION TITLE
SOAP Implementation Issues for ONVIF (To/Action/Dynamic Endpoint)

### DISCUSSION THREAD

**Discussion by Arbane** [CoreWCF/CoreWCF#582] [Q&A]
### SOAP Implementation Issues in ONVIF (To/Action/Dynamic Endpoint)

I'm solving the problem of implementing communication over the [ONVIF](https://www.onvif.org/profiles/specifications/) protocol (this is SOAP). I compiled the current CoreWCF (0.5.0) code for my program.

Most of the ONVIF methods assume that messages do not have a Header. How can I disable (avoid) the "To" and "Action" **headers strong checks**?
I disabled the "To" check by specifying Soap12
```
    var message = new TextMessageEncodingBindingElement(MessageVersion.Soap12, Encoding.UTF8);
    var transport = new HttpTransportBindingElement();
    var noAuthNoActionBinding = new CustomBinding(message, transport)
    {
        Name = "NoAuthNoActionBinding"
    };
```
However, I can't get rid of the "Action" check in a simple way. When the Action is empty, I just have to use the first tag in the Body to define the method. I only see the way to edit the CoreWCF sources. Is there an easy way?

For ONVIF, I need to **dynamically create Endpoints** and serve them. As I understand it, there is a difference between the recommended methods for the Framework and for implementation in CoreWCF. Please tell me how can I do this.

**Comment by mconnew** [comment]
Do you have this working on .NET Framework or is this a new application? If you've got this working on .NET Framework, then the easiest way is to translate what you're doing there to CoreWCF. Otherwise we can get creative and come up with ways to achieve this. I think a custom binding element which modifies the message might work. Or maybe use MessageVersion.None.
